### ISSUE TITLE
Extremely slow for training Flownet2C

### ISSUE THREAD

**Issue by lliuz** [NVIDIA/flownet2-pytorch#15]
I'm trying to train FlowNet2 separately. It's very slow when I train the flownet2C, around 22s/batch with batch_size=32 on 2 P40 GPUs, while flownet2S takes 1.5s/batch. I don't know if there any problems, I want to know how fast you train FlowNet2C or FlowNet2

**Comment by lliuz** [comment]
My runtime environment is based on the `nvidia/cuda:8.0-cudnn5-devel-ubuntu16.04` docker image, with pytorch 0.3.0.post4

**Comment by lliuz** [comment]
The forward is fast(0.6s/batch), I think the problem is in backward

**Comment by PkuRainBow** [comment]
I am training the Flownet-S and the batch size is 8, the time is about 1.58s/batch. I am using TITAN X(Pascal)

**Comment by fitsumreda** [comment]
@PkuRainBow does the 1.58s/batch include both forward and backward passes?

**Comment by PkuRainBow** [comment]
@fitsumreda I am not very sure whether the time includes forward & backward as it is shown as in your code like this:
![image](https://user-images.githubusercontent.com/4639578/34465449-1e69305a-eee8-11e7-866c-eda0e6683fd5.png)
So does "1.59s/it" include forward & backward or other meaning?

**Comment by fitsumreda** [comment]
@PkuRainBow Yes, it includes both forward and backward passes for the specified batch_size. This speed on Titan X is within expectations. I'll close this issue.

**Comment by PkuRainBow** [comment]
@fitsumreda If so, we need to train 1000 epoch, every epoch takes 22,000/batch size iterations. So if the batch size is 8, then epoch takes 3000 iterations, so we will train for 1000 x 3000 = 3,000,000 iterations. Then the overall time cost based on a single TITAN X would be 3,000,000 x 1.6s = 55.5 days!!!  Amazing huge time cost! So could you share me how to accelerate the training procedure.

**Comment by MatthewD1993** [comment]
@lliuz @PkuRainBow Hey, guys. Do you get a reasonable result when training the FlowNet2C? My training curve is still very noisy after 5 epochs, and when I visualize the flow image of a small test set, seems it does not learn anything. In contrast to FlowNetS, which can be trained very fast, and predict resonable initial flow image after 1 epoch. Could you share how you train the FlowNet2C and the key factors to make it work? Thanks!

**Comment by qiuzhongwei-USTB** [comment]
@lliuz hi, have you trained flownet2 successfully? could you tell me how freeze parameter?

**Comment by zhanghaozhou** [comment]
@PkuRainBow Hi, I see the image you upload in this channel. I am also training FlowNetC and I found your training results really interesting at epoch 189. Are you just using these codes and FlyingChairs dataset to train the model?
